manifest/oci_index.go

```diff
@@ -89,16 +98,45 @@ func (index *OCI1Index) ChooseInstance(ctx *types.SystemContext) (digest.Digest,
 				Variant:      d.Platform.Variant,
 			}
 			if platform.MatchesPlatform(imagePlatform, wantedPlatform) {
-				return d.Digest, nil
+				if matchedDigest == "" {
+					matchedDigest = d.Digest
+				}
+				// Always give priority to zstd if more than one image
+				// exists for the same platform.
+				// TODO: In future this check will be configurable and
+				// will be relayed via `ctx`.
+				if _, ok := d.Annotations[OCI1InstanceAnnotationCompressionZSTD]; ok {
+					matchedDigest = d.Digest
+					return matchedDigest, nil
+				}
 			}
 		}
 	}
 
+	if matchedDigest != "" {
+		return matchedDigest, nil
+	}
+
 	for _, d := range index.Manifests {
 		if d.Platform == nil {
-			return d.Digest, nil
+			if matchedDigest == "" {
+				matchedDigest = d.Digest
+			}
+			// Always give priority to zstd if more than one image
+			// exists for the same platform.
+			// TODO: In future this check will be configurable and
+			// will be relayed via `ctx`.
```

Comment:
I am not thrilled about delaying making it conditional.  There's a risk this commit could make it into a release and that may break someone on something somewhere.

Reply:
Made it configurable in a new commit.